lightning/src/chain/package.rs

```diff
@@ -646,16 +649,11 @@ impl PackageTemplate {
 		// If old feerate is 0, first iteration of this claim, use normal fee calculation
 		if self.feerate_previous != 0 {
 			if let Some((new_fee, feerate)) = feerate_bump(predicted_weight, input_amounts, self.feerate_previous, fee_estimator, logger) {
-				// If new computed fee is superior at the whole claimable amount burn all in fees
-				if new_fee > input_amounts {
-					return Some((0, feerate));
-				} else {
-					return Some((input_amounts - new_fee, feerate));
-				}
+				return Some((cmp::max(input_amounts as i64 - new_fee as i64, dust_limit_sats as i64) as u64, feerate));
```

Comment:
I think we'll want to use `u64` for the `cmp::max` parameters. Otherwise, if `dust_limit_sats` is `u64::max_value`, then casting it to `i64` will be `-1`.

Reply:
Hmm, but the first parameter can go negative, right? Or I'd assume it may. I added an assertion because that feels like a logic error - we'd be trying to spend to a script that the user provided that rust-bitcoin is telling us has an infinite dust limit (which I'd think implies "not ever broadcastable").

Reply:
Ah, right that makes sense. Should we check this earlier when calling `get_destination_script` during channel creation?

Reply:
I mean maybe but adding dead code for this seems like a waste? rust-bitcoin currently only ever returns values that are within reason, so if we want a standardness check we should just...do that?

Reply:
Gotcha... I guess the way I read the discussion in https://github.com/rust-bitcoin/rust-bitcoin/pull/636 was that it may return infinity in the future. Did I misread that?

Anyhow, I'm sure this is fine as is for now.

Reply:
Yea, I mean it may return infinity in the future, that is true, or maybe `None` or...but it seems weird to try to predict what it'll do in our own code and then handle one really narrow case of non-standard outputs and not others?